### ISSUE TITLE
Show menu of non-GNUstep applications

### ISSUE THREAD

**Issue by probonopd** [BertrandDekoninck/TopBar#1]
So the intention of TopBar is to show something when a non-GNUstep application is in the foreground.

Great! :+1: 

Wouldn't it be **super cool** if it would actually show the icons of those applications, in the same style as they are shown by GNUstep for GNUstep applications?

Qt applications on Linux automatically export their menus via dbus, so that global menubars can render them.

Kai Uwe Broulik has written a translator that makes Gtk apps and even Audacity work too!
https://blog.broulik.de/2018/03/gtk-global-menu/

With this, global menus work e.g., in Qt4 apps, Qt4 apps, KDE apps, Gtk2 apps (via `appmenu-gtk-module`), Gtk3 apps, Chrome/Chromium, LibreOffice, Firefox (via `firefox-appmenu-bin`), Java applications like JetBrains IDEs. Basically _all_ apps!

Examples: https://github.com/probonopd/hello/issues/11 and http://www.webupd8.org/2017/02/alternative-global-menu-for-mate-and.html.

It would be awesome if TopBar could implement showing those menus that applications export via dbus.

Here are some implementations in other languages, I didn't find one in ObjC yet:

* https://github.com/jiqing112/Gnome-Global-AppMenu
* https://gitlab.com/vala-panel-project/vala-panel-appmenu
* https://github.com/psifidotos/applet-window-appmenu

**Comment by probonopd** [comment]
I case you worry about having to run `dbus-daemon` (which is running on most Linux and some *BSD systems anyway), it may be possible to just implement the minimum viable portions of it in order to get this working. But I'd say this is something we can worry about once we have something working at all.

**Comment by probonopd** [comment]
While we are at it, please consider adding global search in menus, an absolute **killer feature** (imho) shown here by @afiestas: https://youtu.be/9lGVy6GRbOc?t=140

Maybe integrate a search right into the TopBar similar to what the Mac does (although they unfortunately hide it away in the Help menu, and don't offer a convenient shortcut to bring it up as far as I know):

![image](https://user-images.githubusercontent.com/2480569/93707773-b3fea200-fb31-11ea-86ce-b541e1706cc7.png)

More about this in my article about [menu usability](https://medium.com/@probonopd/make-it-simple-linux-desktop-usability-part-1-5fa0fb369b42).

**Comment by BertrandDekoninck** [comment]
Of course it would be great !
For me, the first step would have been to be able  to catch gnustep menus in the way EtoileMenuServer did. But this only little thing is way beyond my programming kills.
For now, each GNUstep app shows its own menu on TopBar with a transparent theme for gnustep menubar.

So catching gtk or Qt menus and icons is another story.

My TopBar app is only a little rectangle window I put at the top of the screen with very little functionality. I coded it because EtoileMenuServer didn't build anymore with recent gnustep.

**Comment by probonopd** [comment]
Yes, I wonder what it would take to resurrect Etoile and its MenuServer. Do you know why it doesn't build anymore?

**Comment by BertrandDekoninck** [comment]
In fact, EtoileMenuServer did build and run the last time I tried (more than 3 years ago), but it didn't display any menu. Maybe it used a component called EtoileWildMenu, which was created when GNUstep couldn't display horizontal menus. This is now incompatible with GNUstep menu system. The theme system has also been implemented in GNUstep and Etoile's one's (Chamaelon) is now deprecated.
Many components of Etoile don't build anymore. I don't remember if I could build EtoileFoundation

But I've already seen some screenshots of Alessandro Sangiuliano (https://github.com/AlessandroSangiuliano) with EtoileMenuServer running. Maybe he has modified it for its own use.

**Comment by probonopd** [comment]
Seems like @ngrewe already had this idea and had started working on it but unfortunately there is not much code there: https://github.com/gnustep/libs-dbuskit/tree/master/Examples/GlobalMenu.

In any case, maybe https://github.com/gnustep/libs-dbuskit can be helpful to implement this. Or mabe @ngrewe, inspired by this ticket, wants to complete that example? That, of course, would be awesome :+1:

**Comment by ngrewe** [comment]
Just to clarify: The referenced example is a minimal theme that utilises the DBusMenu bundle to delegate displaying the menu to the global menu in Ubuntu instead of rendering it via GNUstep widgets. So you could say that it's implementing the menu _client_ rather than the menu server.

**Comment by probonopd** [comment]
Thanks for chiming in @ngrewe. Ah right, I had confused it. What we are after is being able to show the menus of non-GNUstep applications in a GNUstep global menu, so that they blend more nicely into a GNUstep based desktop.

**Comment by probonopd** [comment]
Hi @BertrandDekoninck could you please enable GitHub Issues on https://github.com/BertrandDekoninck/Gworkspace? I'd have a suggestion there. Thanks a lot.

**Comment by BertrandDekoninck** [comment]
I do not recommand using my fork of GWorkspace for development. It's out of sync with  the offical repo (almost 2 years old) and the historic of commits is borched as I notice in the "About" section of the repo.

You should post them against master at https://github.com/gnustep/apps-gworkspace and deal with Ricardo Mottola for PRs.

**Comment by probonopd** [comment]
Thanks @BertrandDekoninck
